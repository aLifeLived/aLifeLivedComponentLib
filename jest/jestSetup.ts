/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
