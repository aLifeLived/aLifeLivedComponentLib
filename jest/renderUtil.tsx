import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
