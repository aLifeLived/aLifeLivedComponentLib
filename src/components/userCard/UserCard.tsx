import React from 'react';
import { Avatar } from '../avatar/Avatar';
import { Box } from '../../themes/box/Box';
import { SystemButton } from '../../themes/systemButton/SystemButton';
import { Text } from '../../themes/text/Text';

type UserCardProps = {
  username: string;
  avatarUrl: string;
  onPress: () => void;
};

export const UserCard: React.FC<UserCardProps> = ({
  username,
  avatarUrl,
  onPress,
}) => {
  return (
    <SystemButton onPress={() => onPress()}>
      <Box flexDirection="row" alignItems="center">
        <Avatar
          iconBorderRadius="xXXlarge"
          iconHeight={50}
          iconWidth={50}
          source={{ uri: avatarUrl, priority: 'high' }}
          variant="roundAvatar"
        />
        <Box marginLeft="s" width="80%">
          <Text variant="h3" fontWeight="bold" numberOfLines={1}>
            {username}
          </Text>
        </Box>
      </Box>
    </SystemButton>
  );
};
