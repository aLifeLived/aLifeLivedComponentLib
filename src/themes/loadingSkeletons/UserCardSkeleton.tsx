import React from 'react';

// Components
import { Box } from '../box/Box';
import { SkeletonElement } from './SkeletonElement';
import { SystemAnimatableBox } from '../systemAnimatableBox/SystemAnimatableBox';

export const UserCardSkeleton = () => {
  return (
    <SystemAnimatableBox
      flexDirection="row"
      alignContent="center"
      alignItems="center"
      animation="flash"
      iterationCount="infinite"
      useNativeDriver={true}
    >
      <SkeletonElement variant="avatar" />
      <Box flexDirection="column" width="100%" marginLeft="s">
        <SkeletonElement variant="title" />
        <SkeletonElement variant="text" />
      </Box>
    </SystemAnimatableBox>
  );
};
