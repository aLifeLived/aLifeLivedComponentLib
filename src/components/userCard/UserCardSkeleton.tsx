import React from 'react';

// Components
import { Box } from '../../themes/box/Box';
import { SkeletonElement } from '../../themes/loadingSkeletons/SkeletonElement';
import { SystemAnimatableBox } from '../../themes/systemAnimatableBox/SystemAnimatableBox';

type UserCardSkeleton = {
  testID?: string;
};

export const UserCardSkeleton: React.FC<UserCardSkeleton> = ({ testID }) => {
  return (
    <SystemAnimatableBox
      flexDirection="row"
      animation="flash"
      iterationCount="infinite"
      alignItems="center"
      useNativeDriver={true}
    >
      <SkeletonElement variant="avatar" />
      <Box width="100%" marginLeft="s" testID={testID}>
        <SkeletonElement variant="title" />
      </Box>
    </SystemAnimatableBox>
  );
};
