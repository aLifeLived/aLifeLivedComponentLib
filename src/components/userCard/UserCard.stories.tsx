import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

// Component
import { UserCard } from './UserCard';
import { UserCardSkeleton } from './UserCardSkeleton';
import { Box } from '../../themes/box/Box';
import { SystemFlatList } from '../../themes/systemFlatList/SystemFlatList';

const data = [
  {
    id: '5f92cdce0cf217478ba93163',
    username: 'johnSmithWithAReallyReallyLongNameSoLOng',
    avatarImageMediaAsset: {
      url: 'https://picsum.photos/id/237/200/300',
    },
  },
  {
    id: '5f92cdce0cf217478ba93263',
    username: 'johnSmith',
    avatarImageMediaAsset: undefined,
  },
  {
    id: '5f92cdce0cf217478ba93663',
    username: 'johnSmith',
    avatarImageMediaAsset: {
      url: 'https://picsum.photos/id/237/200/300',
    },
  },
  {
    id: '5f92cdce0cf217478ba93561',
    username: 'johnSmith',
    avatarImageMediaAsset: {
      url: '',
    },
  },
];

storiesOf('User card', module)
  .addDecorator(Story => Story())
  .add('Default user card', () => (
    <Box
      backgroundColor="surface"
      flex={1}
      alignContent="center"
      alignItems="flex-start"
    >
      <SystemFlatList
        data={data}
        width="100%"
        keyExtractor={item => item.id}
        renderItem={({ item }) => (
          <Box marginTop="m" width="100%" paddingLeft="s">
            <UserCard
              onPress={action('On press fired')}
              username={item.username}
              avatarUrl={item.avatarImageMediaAsset?.url}
            />
          </Box>
        )}
      />
    </Box>
  ))
  .add('User card loading', () => (
    <Box margin="m">
      <UserCardSkeleton />
    </Box>
  ));
