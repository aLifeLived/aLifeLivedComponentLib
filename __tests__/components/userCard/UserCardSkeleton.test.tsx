import React from 'react';
import { render } from '../../../jest/renderUtil';

// Component
import { UserCardSkeleton } from '../../../src/components/userCard/UserCardSkeleton';

const setup = ({ testID }: { testID?: string }) => {
  const utils = render(<UserCardSkeleton testID={testID} />);

  return {
    ...utils,
  };
};

describe('themes >> Loading Skeleton >> UserCardSkeleton', () => {
  describe('Given User card', () => {
    test('Should match snapshot', () => {
      const { toJSON } = setup({});
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Given user card skeleton has a test id', () => {
    test('Should find testID', () => {
      const { getByTestId } = setup({ testID: 'user-card-skeleton-id' });
      expect(getByTestId('user-card-skeleton-id')).toBeDefined();
    });
  });
});
