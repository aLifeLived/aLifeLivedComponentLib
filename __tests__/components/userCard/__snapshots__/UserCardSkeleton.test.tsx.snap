// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> Loading Skeleton >> UserCardSkeleton Given User card Should match snapshot 1`] = `
<View
  forwardedRef={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
        Array [
          Object {},
          Array [
            Object {
              "borderColor": "#CBCBCB",
              "borderRadius": 48,
              "borderWidth": 1,
              "height": 50,
              "width": 50,
            },
          ],
        ],
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "marginLeft": 8,
          "width": "100%",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
          },
          Array [
            Object {},
            Array [
              Object {
                "backgroundColor": "#CBCBCB",
                "height": 20,
                "width": "50%",
              },
            ],
          ],
        ]
      }
    />
  </View>
</View>
`;
